phylanx/ir/node_data.hpp

```diff
@@ -155,13 +155,13 @@ namespace phylanx { namespace ir
             dimensions_type dimensions_;
 
             // support for intrusive_ptr
-            template <typename T>
-            friend void intrusive_ptr_add_ref(node_data_storage_base<T>* p)
+            template <typename _T>
```

Comment:
Please don't use leading underscores in identifiers. Those names are reserved for the implementation. Could you use `T_` instead, for instance, please?